### PR TITLE
BUG: Fix RecursionError when using a scalar point to select IntervalIndex columns

### LINKED ISSUE #26490: RecursionError when selecting single column from IntervalIndex
#### Code Sample, a copy-pastable example if possible

```python
df = pd.DataFrame(
        np.ones((3, 4)), 
        columns=pd.IntervalIndex.from_breaks(np.arange(5)))
df[0.5]
df.loc[:, 0.5]
```
#### Problem description

Instead of returning the selected column, either calling `df[0.5]` or `df.loc[:, 0.5]` raises `RecursionError`.

The issue is in `frame.__getitem__` where `key in self.columns == False`. The code then correctly identifies the desired column by integer index using `self.columns.get_loc(key)`, but then goes on to call 
```python
data = self._take(indexer, axis=1)
data = data[key]
```
Because `self._take` returns a DataFrame, `data[key]` enters `frame.__getitem__` again and we are caught in an infinite loop.

#### Expected Output

The selected column as Series.

#### Output of ``pd.show_versions()``

<details>

commit: None
python: 3.6.8.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.24.2
pytest: 4.3.1
pip: 19.0.3
setuptools: 40.8.0
Cython: 0.29.7
numpy: 1.16.2
scipy: 1.2.1
pyarrow: None
xarray: None
IPython: 7.4.0
sphinx: 1.8.5
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: None
tables: 3.5.1
numexpr: 2.6.9
feather: None
matplotlib: 3.0.3
openpyxl: 2.6.1
xlrd: 1.2.0
xlwt: None
xlsxwriter: None
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: 1.3.1
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by jschendel**
- [X] closes #26490
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2866,7 +2866,12 @@ def __getitem__(self, key):
         if is_single_key:
             if self.columns.nlevels > 1:
                 return self._getitem_multilevel(key)
+
             indexer = self.columns.get_loc(key)
+            if isinstance(self.columns, ABCIntervalIndex):
```
Comment (jreback) [inline pandas/core/frame.py]: rather would not inject specific index logic here

you can make this not a single key (so it takes a different branch)

or define one of the _convert_to_index routines which dispatch to the index type

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@jschendel can you merge master and had some comments

**Comment by jbrockmendel** [comment]
@jschendel can you rebase

**Comment by jschendel** [comment]
I don't have a cleaner implementation here, and will need to come back to this later.  Closing for now as I don't want this sitting around stale.

**Pull request closed without merging**